.github/workflows/backport-queue.yml

```diff
@@ -0,0 +1,169 @@
+          - { release-line: v22.x, base-branch: v23.x }
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
+        with:
+          # Needs the whole git history for ncu to work
+          # See https://github.com/nodejs/node-core-utils/pull/486
+          fetch-depth: 0
+          ref: ${{ matrix.branch.release-line }}-staging
+          # A personal token is required because pushing changes to `.github`
+          # with default token will be blocked. It needs
+          # to be set here because `checkout` configures GitHub authentication
+          # for push as well.
+          token: ${{ secrets.GH_USER_TOKEN }}
+
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b  # v5.3.0
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      # Install dependencies
+      - name: Install Node.js
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af  # v4.1.0
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Install branch-diff
+        run: npm install -g branch-diff
+
+      - name: Setup git author
+        run: |
+          git config --local user.email "github-bot@iojs.org"
+          git config --local user.name "Node.js GitHub Bot"
+
+      - name: Set up ghauth config (Ubuntu)
+        run: |
+          mkdir -p "${XDG_CONFIG_HOME:-~/.config}/changelog-maker"
+          echo '{}' | jq '{user: env.USERNAME, token: env.TOKEN}' > "${XDG_CONFIG_HOME:-~/.config}/changelog-maker/config.json"
+        env:
+          USERNAME: ${{ secrets.JENKINS_USER }}
+          TOKEN: ${{ github.token }}
+
+      - name: Fetch base branch
+        run: |
+          git remote set-branches --add origin "$BASE_BRANCH"
+          git fetch origin "$BASE_BRANCH"
+        env:
+          BASE_BRANCH: ${{ matrix.branch.base-branch }}
+
+      - name: Fetch auto-backport branch if it exists
+        id: fetch
+        run: |
+          if git fetch origin "$BACKPORT_BRANCH"; then
+            STAGING_BRANCH_TIP="$(git rev-parse HEAD)"
+            WORKING_BRANCH_TIP="$(git rev-parse FETCH_HEAD)"
+            echo "WORKING_BRANCH_TIP=$WORKING_BRANCH_TIP" >> "$GITHUB_OUTPUT"
+            if [ "$WORKING_BRANCH_TIP" != "$STAGING_BRANCH_TIP" ]; then
+                git reset  "$WORKING_BRANCH_TIP" --hard
+                git rebase "$STAGING_BRANCH_TIP" --empty=drop || git reset "$STAGING_BRANCH_TIP" --hard
+            fi
+          else
+            echo "Branch doesn't exist yet"
+          fi
+        env:
+          BACKPORT_BRANCH: ${{ matrix.branch.release-line }}-staging-auto-backport
+
+      - name: Run the queue
+        id: queue
+        run: |
+          set -xe
+
+          NODE="$(command -v node)"
+
+          backport() {
+            git cherry-pick "$1" && \
+            make lint-js NODE="$NODE" && \
+            make test-doc -j4 NODE="$NODE"
```

Comment:
The more I think about this problem space, the more I'm convinced that the right way to go about it is to complete the backport queue ignoring tests and automate a `git bisect` workflow that detects **when** things broke only **if** we actually have a broken build in the staging branch.

Reply:
Running doc tests is quite cheap, I don't see a reason not to

Reply:
sure thing, that could easily be incorporated into https://github.com/nodejs/node-core-utils/pull/875 too 👍 

I'm just making the case that running the full test suite is the most valuable thing and that is very expensive and because of that I've been thinking more and more that bisecting later might actually be the way to go, even for automation.